Grammar/python.gram

```diff
@@ -1273,6 +1273,19 @@ invalid_named_expression(memo):
                                           _PyPegen_get_expr_name(a)) }
 
 invalid_assignment:
+    # 'yield' is handled below
+    | a[Token*]=('False' | 'None' | 'True' | 'and' | 'as' | 'assert' | 'async'
```

Comment:
I am not super happy listing all the keywords here. I am a bit more amenable since this is an invalid rule but the list of keywords are automatically generated and he have functions to check if a name token is a keyword for that reason.

There are two possibilities: we can add a special meta token that matches any keyword and then use that or we can match NAME and filter in the action.

Reply:
@sobolevn great minds think alike https://github.com/python/cpython/pull/138888#discussion_r2347289485